gui/src/components/SuiteForm.js

```diff
@@ -0,0 +1,55 @@
+import {useRef} from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+
+function SuiteForm({socket}) {
+  const formRef = useRef();
+  const tests = [{
+      name: "test-test",
+      mapKey: "testTest",
+      label: "Test Test"
+    }, {
+      name: "test-test-duplicate",
+      mapKey: "testTestDuplicate",
+      label: "Test Test Duplicate"
+    }
+  ];
+
+  const sendSuite = (event) => {
+    event.preventDefault();
+    const testStr = tests.filter(
+      test => formRef.current[test.name].checked
+    ).map(
+      test => `"${test.mapKey}": {}`
+    ).join();
+    const msg = `{
+      "message-type": "CREATE-SUITE",
+      "url": "${formRef.current["url"].value}",
+      "tests": {${testStr}}
+    }`;
+    console.log(`Message to send: ${msg}`);
+    console.log(`Socket state: ${socket.readyState}`);
+    if (socket.readyState === 1) socket.send(msg);
+  }
+
+  return (
+    <Form ref={formRef} className="w-25 py-2 px-4 bg-secondary text-white d-flex flex-column justify-content-between">
+      <Form.Group>
+        <Form.Label>URL</Form.Label>
+        <Form.Control name="url" type="text" placeholder="google.com"/>
```

Comment:
You need to pass the protocol otherwise the request will fail, update to https://google.com

Reply:
yes, I'm not sure If the tests should prepend that or not, what do you think?

Reply:
The way I see if we have 2 choices
- Make users append `https://` to all the functions
- Added that into Daniels web request function al la f"https://{url}" 

The latter might be a better way but maybe we should discuss it tonight

Reply:
Agreed